diff -up -r mn_search.context.inc mn_search.context.inc
--- mn_search.context.inc	2013-07-19 12:43:23.751798913 +0200
+++ mn_search.context.inc	2013-07-19 12:47:01.957094503 +0200
@@ -1,7 +1,7 @@
 <?php
 
 /**
- * Implementation of hook_context_default_contexts().
+ * Implements hook_context_default_contexts().
  */
 function mn_search_context_default_contexts() {
   $export = array();
diff -up -r mn_search.features.inc mn_search.features.inc
--- mn_search.features.inc	2013-07-19 12:43:23.751798913 +0200
+++ mn_search.features.inc	2013-07-19 12:47:01.965094549 +0200
@@ -1,7 +1,7 @@
 <?php
 
 /**
- * Implementation of hook_ctools_plugin_api().
+ * Implements hook_ctools_plugin_api().
  */
 function mn_search_ctools_plugin_api() {
   list($module, $api) = func_get_args();
@@ -11,7 +11,7 @@ function mn_search_ctools_plugin_api() {
 }
 
 /**
- * Implementation of hook_views_api().
+ * Implements hook_views_api().
  */
 function mn_search_views_api() {
   return array(
diff -up -r mn_search.info mn_search.info
--- mn_search.info	2013-07-19 12:43:23.763798985 +0200
+++ mn_search.info	2013-07-19 12:47:02.205095974 +0200
@@ -1,4 +1,4 @@
-core = "6.x"
+core = 7.x
 dependencies[] = "context"
 dependencies[] = "flot"
 dependencies[] = "geotaxonomy"
@@ -17,3 +17,4 @@ features[views_api][] = "api:2"
 name = "MN Search"
 package = "Managing News"
 project = "mn_search"
+
diff -up -r mn_search.module mn_search.module
--- mn_search.module	2013-07-19 12:43:23.867799602 +0200
+++ mn_search.module	2013-07-19 12:47:02.077095214 +0200
@@ -1,6 +1,6 @@
 <?php
 
-include_once('mn_search.features.inc');
+include_once DRUPAL_ROOT . '/' . 'mn_search.features.inc';
 
 /**
  * Clear search status cache.
@@ -14,7 +14,7 @@ function mn_search_cron() {
  */
 function mn_search_get_search_status($reset = FALSE) {
   $cache = cache_get('mn_search_status');
-  if ($reset || !$cache || (!empty($cache->data['expires']) && time() > $cache->data['expires'])) {
+  if ($reset || !$cache || (!empty($cache->data['expires']) && REQUEST_TIME > $cache->data['expires'])) {
     // Only invoke data_search... we don't care about nodes, users, etc.
     $status = module_invoke('data_search', 'search', 'status');
     $status['expires'] = strtotime('+1 hours');
@@ -25,40 +25,46 @@ function mn_search_get_search_status($re
 }
 
 /**
- * Implementation of hook_block().
+ * Implements hook_block_info().
  */
-function mn_search_block($op = 'list', $delta = 0) {
-  switch ($op) {
-    case 'list':
-      $blocks = array();
-      $blocks['status']['info'] = t('Search index status');
-      return $blocks;
-    case 'view':
-      switch ($delta) {
-        case 'status':
-          if (user_access('administer mn')) {
-            $status = mn_search_get_search_status();
-            $remaining = isset($status['remaining']) ? $status['remaining'] : 0;
-            $total = isset($status['total']) ? $status['total'] : 0;
-
-            // Only show the block if there are items to be indexed.
-            if ($remaining > 0) {
-              $percentage = ((int) min(100, 100 * ($total - $remaining) / max(1, $total))) . '%';
-              $label = t('Search index');
-              $suffix = t('complete');
-              $output = "<label>{$label}</label> <strong>{$percentage}</strong> {$suffix}";
-              $output .= "<div class='description'>" . l(t('Update now'), 'admin/reports/status/run-cron', array('query' => drupal_get_destination())) . "</div>";
-              return array('content' => $output);
-            }
-          }
-          break;
+function mn_search_block_info() {
+  $blocks = array();
+  $blocks['status']['info'] = t('Search index status');
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_view().
+ */
+function mn_search_block_view($delta) {
+  switch ($delta) {
+    case 'status':
+      if (user_access('administer mn')) {
+        $status = mn_search_get_search_status();
+        $remaining = isset($status['remaining']) ? $status['remaining'] : 0;
+        $total = isset($status['total']) ? $status['total'] : 0;
+
+        // Only show the block if there are items to be indexed.
+        if ($remaining > 0) {
+          $percentage = ((int) min(100, 100 * ($total - $remaining) / max(1, $total))) . '%';
+          $label = t('Search index');
+          $suffix = t('complete');
+          $output = "<label>{$label}</label> <strong>{$percentage}</strong> {$suffix}";
+          $output .= "<div class='description'>" . l(t('Update now'), 'admin/reports/status/run-cron', array('query' => drupal_get_destination())) . "</div>";
+          return array('content' => $output);
+        }
       }
       break;
   }
 }
 
 /**
- * Implementation of hook_help().
+ * Implements hook_block().
+ */
+function mn_search_block_OLD($op = 'list', $delta = 0) { }
+
+/**
+ * Implements hook_help().
  */
 function mn_search_help($path, $arg) {
   switch ($path) {
@@ -77,7 +83,7 @@ function mn_search_help($path, $arg) {
 }
 
 /**
- * Implementation of hook_menu_alter().
+ * Implements hook_menu_alter().
  *
  * Hiding search local tasks ('content', 'user', etc.)
  */
@@ -90,7 +96,7 @@ function mn_search_menu_alter(&$items) {
 }
 
 /**
- * Implementation of hook_form_alter() for views_exposed_form.
+ * Implements hook_form_alter() for views_exposed_form().
  *
  * Alter the filter button label to say "Search"
  */
@@ -117,18 +123,22 @@ function mn_search_form_stored_views_for
 }
 
 /**
- * Implementation of hook_views_pre_render().
+ * Implements hook_views_pre_render().
  */
 function mn_search_views_pre_render(&$view) {
   if ($view->display_handler->display->display_plugin === 'page') {
     $input = $view->get_exposed_input();
     if (!empty($input['search'])) {
-      drupal_set_title(check_plain($input['search']));
+      drupal_set_title($input['search']);
     }
   }
 }
 
 
+/**
+ * @todo Please document this function.
+ * @see http://drupal.org/node/1354
+ */
 function mn_search_openlayers_behaviors_info() {
   $info = array();
 
@@ -154,7 +164,7 @@ function mn_search_openlayers_behaviors_
  *   Behavior array
  */
 function mn_search_behaviors_process_semi_fullscreen($behavior, &$map) {
-  drupal_add_css(drupal_get_path('module', 'mn_search') . '/mn_search.css', 'module');
+  drupal_add_css(drupal_get_path('module', 'mn_search') . '/mn_search.css');
   return $behavior;
 }
 
diff -up -r mn_search.views_default.inc mn_search.views_default.inc
--- mn_search.views_default.inc	2013-07-19 12:43:23.867799602 +0200
+++ mn_search.views_default.inc	2013-07-19 12:47:02.205095974 +0200
@@ -1,7 +1,7 @@
 <?php
 
 /**
- * Implementation of hook_views_default_views().
+ * Implements hook_views_default_views().
  */
 function mn_search_views_default_views() {
   $views = array();
